sql/parser/builtins.go

```diff
@@ -48,9 +48,15 @@ var errRoundNumberDigits = errors.New("number of digits must be greater than 0")
 type typeList []reflect.Type
 
 type builtin struct {
-	// Set to typeList{} for nullary functions and to nil for varidic
-	// functions.
+	// Variadic functions have a tricky representation:
+	// - If all arguments must be of a specific type, `types` must be
+	// `typeList{<theType>}`, and `variadic` must be true. This will
+	// generate the correct type-checking error.
+	// - If all arguments must be of the same (variable) type, `types`
```

Comment:
Yeah, I'm open to rewording suggestions; that `variadic == true` is not used is what I mean when I say `This representation is not quite "normal", sorry about that.` in the comment